doc/source/data/data.rst

```diff
@@ -66,22 +78,6 @@ Learn more
 
             Learn how to use Ray Data
 
-    .. grid-item-card::
-
-        **Key Concepts**
```

Comment:
Can we update the description in the Gettin started card now that it is more of a Key Concepts page?